### PR TITLE
Implement backend and frontend for shardstatus

### LINKED ISSUE #104: Show shard status on website
Displaying the status of the shards, connected users, guilds etc on the dashboard would be cool

### PR DESCRIPTION
**Pull request by OtterBoops**
Fixes #104

### CODE REVIEW

File: dashboard/src/components/status/Status.vue [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  - along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<template>
+	<div class="Status">
+		I am inevitable
+
+		<div class="shard">
+			<p>{{ shard.id }}</p>
+			<p>{{ shard.status }}</p>
```
Comment (OtterBoops) [inline dashboard/src/components/status/Status.vue:24]: For some reason shard id and status return undefined.
I'll need some help to figure out why.

File: dashboard/src/components/status/shard-store.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+const FETCH_SHARDS_INTERNAL = "FETCH_SHARDS_INTERNAL";
+
+const LOAD_SHARDS = "LOAD_SHARDS";
+const FETCHING_SHARDS = "FETCHING_SHARDS";
+
+export const shardStore = {
+	namespaced: true,
+	modules: {},
+	state: () => ({
+		shardsLoading: false, //true each time there is a request for staff in flight
```
Comment (napstr) [inline dashboard/src/components/status/shard-store.js:31]: solid copy paste development but you missed the "staff" comment 🙃

Comment (OtterBoops) [reply dashboard/src/components/status/shard-store.js]: Oops.

**Comment by OtterBoops** [comment]
As you've said, a lot of the thing is copy-pasted. Let me know if I've carried over too much or too little. The only issue that is blocking me right now is Front-End consuming the shard info.

**Comment by napstr** [comment]
> Let me know if I've carried over too much or too little.
All good so far. A bit of duplicated code is much cheaper than creating the wrong abstraction.

I probably won't be able to do an in-depth review of this until tomorrow evening.

File: dashboard/src/components/status/Status.vue [hunk 3]
```diff
@@ -19,7 +19,12 @@
 	<div class="Status">
     <div class="is-size-1">Wolfia Shard Status</div>
 		<div class="stafflist columns is-centered is-multiline" :class="{ 'is-loading': !shardsLoaded }">
-      <Shard />
+      <div
+        v-for="shard in shards"
+				:key="shard"
```
Comment (napstr) [inline dashboard/src/components/status/Status.vue:24]: This should probably be the shard ID
I can't really explain what it does exactly but the Vue docs are insisting that its important: https://vuejs.org/v2/guide/list.html#Maintaining-State

**Comment by OtterBoops** [comment]
The only changes I have on my mind right now are to make the shards cards and their spacing smaller. Your call.
Any further ideas/suggestions are welcomed.

**Pull request merged**